x-pack/plugins/discover_enhanced/ui_tests/fixtures/index.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  test as base,
+  PageObjects,
+  createLazyPageObject,
+  ScoutTestFixtures,
+  ScoutWorkerFixtures,
+} from '@kbn/scout';
+import { DemoPage } from './page_objects';
+
+interface ExtendedScoutTestFixtures extends ScoutTestFixtures {
+  pageObjects: PageObjects & {
+    demo: DemoPage;
+  };
+}
+
+export const test = base.extend<ExtendedScoutTestFixtures, ScoutWorkerFixtures>({
+  pageObjects: async ({ pageObjects, page }, use) => {
+    const extendedPageObjects = {
+      ...pageObjects,
+      demo: createLazyPageObject(DemoPage, page),
+    };
+
+    await use(extendedPageObjects);
+  },
+});
```

Comment:
So the idea here is that each plugin can customize the base test figures with their own?

I wonder how this would work if say I'm testing Lens but need some dashboard fixtures, I guess just import them too.

Reply:
> I wonder how this would work if say I'm testing Lens but need some dashboard fixtures, I guess just import them too.

core fixtures together with page objects fixture will be available in the context. It solves few problems we have with FTR today:
- page objects/service explicit loading with `getService`
- duplicated services & Page Objects across the repo
- compatibility issues due to multiple `FtrContextProvider` implementations

Scout will export core fixtures and we expect Teams to contribute back in package all the Fixtures and Page Objects that can be used in more than 1 plugin. Both test and worker fixtures are lazy loaded, so there is no issue of having all of it in the test context. In case of need, you can override or extend existing one directly in plugin or contribute back to "core" in package.

Page Object instance is lazy loaded with `createLazyPageObject` so there is no performance degradation of exposing all of platform core UI (dashboard, discover, lens, advanced settings, etc) out of the box. It is only created when you explicitly call it in the test:

```
  test.beforeEach(async ({ pageObjects }) => {
    await pageObjects.discover.goto();
    // only discover PO instance was created
  });
```

We will make sure to document the approach, we have a roadmap and we need some time before we will open it for "GA"